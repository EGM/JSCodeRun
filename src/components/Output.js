import React from 'react';
import { Text, ScrollView, View, StyleSheet, Image, TouchableOpacity } from 'react-native';

const styles = StyleSheet.create({
  container: {
    padding: 10,
    flex: 1,
    flexDirection: 'row',
    borderTopWidth: 5,
    borderColor: '#777'
  },
  output: {
    flex: 15
  },
  trashContainer: {
    flex: 1
  },
  trashImage: {
    tintColor: '#777',
    width: 20,
    height: 20,
    alignSelf: 'flex-end'
  }
});

const successStyle = StyleSheet.create({
  text: {
    color: 'black'
  }
});

const errorStyle = StyleSheet.create({
  text: {
    color: 'darkred'
  }
});

const Output = ({ output, onClearOutput }) => (
  <View style={styles.container}>
    <View style={styles.output}>
      <ScrollView>
        {output && output.length
          ? output.map((line, i) => {
              const styles = line.status === 'ERROR' ? errorStyle : successStyle;
              return (
                <View key={i} style={styles.lineWrapper}>
                  <Text style={styles.text}>{line.message}</Text>
                </View>
              );
            })
          : null}
      </ScrollView>
    </View>
    <View style={styles.trashContainer}>
      <TouchableOpacity onPress={() => onClearOutput()}>
        <Image style={styles.trashImage} source={require('../../assets/delete.png')} />
      </TouchableOpacity>
    </View>
  </View>
);

export default Output;
