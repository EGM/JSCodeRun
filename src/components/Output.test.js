import React, { Component } from 'react';
import renderer from 'react-test-renderer';
import Output from './Output';

describe('Output', () => {
  it('should render the Output correctly', () => {
    const mockOutput = [
      {
        message: 'Hello',
        status: 'OK'
      }, {
        message: 'World!',
        status: 'ERROR'
      }, 
    ]
    const rendered = renderer.create(
      <Output 
        output={mockOutput}
        onClearOutput={() => {}}
      />
    ).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});