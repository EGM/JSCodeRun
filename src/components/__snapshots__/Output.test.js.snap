// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Output should render the Output correctly 1`] = `
<View
  style={
    Object {
      "borderColor": "#777",
      "borderTopWidth": 5,
      "flex": 1,
      "flexDirection": "row",
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "flex": 15,
      }
    }
  >
    <RCTScrollView>
      <View>
        <View
          style={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "black",
              }
            }
          >
            Hello
          </Text>
        </View>
        <View
          style={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "darkred",
              }
            }
          >
            World!
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/delete.png",
          }
        }
        style={
          Object {
            "alignSelf": "flex-end",
            "height": 20,
            "tintColor": "#777",
            "width": 20,
          }
        }
      />
    </View>
  </View>
</View>
`;
